store/store_kv.go

```diff
@@ -0,0 +1,138 @@
+package store
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/portworx/kvdb"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	cloudDriveKey     = "clouddrive/"
+	cloudDriveLockKey = "_lock"
+	maxWatchErrors    = 5
+	defaultLockTryDuration = 1 * time.Minute
+	defaultLockHoldDuration = 3 * time.Minute
+)
+
+var (
+	// ErrKvdbNotInitialized is returned when kvdb is not initialized
+	ErrKvdbNotInitialized = errors.New("KVDB is not initialized")
+)
+
+type kvStore struct {
+	k                kvdb.Kvdb
+	keyName          string
+	lockTryDuration  time.Duration
+	lockHoldDuration time.Duration
+}
+
+// NewKVStore returns a Store implementation which is a wrapper over
+// kvdb.
+func NewKVStore(kv kvdb.Kvdb) (Store, error) {
+	return NewKVStoreWithParams(kv, cloudDriveLockKey, 0, 0)
+}
+
+// NewKVStoreWithParams returns a Store implementation which is a wrapper over
+// kvdb.
+func NewKVStoreWithParams(
+	kv kvdb.Kvdb,
+	name string,
+	lockTryDuration time.Duration,
+	lockHoldDuration time.Duration,
+) (Store, error) {
+	kstore := kvStore{}
+	if kv == nil {
+		return nil, ErrKvdbNotInitialized
```

Comment:
Name of the error seems misleading. It made me think that kvdb has not been setup yet. We could just use InvalidArgument or something like that?

Reply:
This is from the existing code and I feel this it's technically correct to call ErrKvdbNotInitialized since it's an indication that the bootstrap kvdb is not initialized.